extensions/wikia/Recirculation/services/CakeRelatedContentService.php

```diff
@@ -21,105 +25,80 @@ class CakeRelatedContentService {
+			$this->warning( static::ARTICLES_ARE_NOT_ARRAY,
+				array_merge( [ "fandom_articles" => $filteredRelatedContent->getFandomArticles() ], $context ) );
+			$filteredRelatedContent->setFandomArticles( [] );
+		}
 
-			if (!is_array($filteredRelatedContent->getWikiArticles())) {
-				$this->warning("getRelatedContentFromEntityName expected wiki_articles to be an array", [
-						"title" => $title,
-						"limit" => $limit,
-						"wiki_articles" => $filteredRelatedContent->getWikiArticles()
-				]);
-				$filteredRelatedContent->setWikiArticles([]);
-			}
+		if ( !is_array( $filteredRelatedContent->getDiscussionThreads() ) ) {
+			$this->warning( static::THREADS_ARE_NOT_ARRAY,
+				array_merge( [ "discussion_threads" => $filteredRelatedContent->getDiscussionThreads() ], $context ) );
+			$filteredRelatedContent->setDiscussionThreads( [] );
+		}
 
-			$wikiArticles = [];
-			foreach ($filteredRelatedContent->getWikiArticles() as $article) {
-				$parsed = parse_url($article->getContent()->getUrl());
-				if (urldecode($parsed['path']) != $ignore) {
-					$wikiArticles[] = $article;
-				}
-			}
+		if ( !is_array( $filteredRelatedContent->getWikiArticles() ) ) {
+			$this->warning( static::WIKI_ARTICLES_ARE_NOT_ARRAY,
+				array_merge( [ "wiki_articles" => $filteredRelatedContent->getWikiArticles() ], $context ) );
+			$filteredRelatedContent->setWikiArticles( [] );
+		}
 
-			/**
-			 * this seems funky, but http://php.net/manual/en/function.array-map.php#refsect1-function.array-map-examples
-			 * so this actually will create an array of arrays where the elements are ordered:
-			 * [
-			 *   // fandom
-			 *   // discussion
-			 *   // article
-			 * ]
-			 * because that's the order they're passed in
-			 */
-			$ordered = array_map(
-					null,
-					$filteredRelatedContent->getFandomArticles(),
-					$filteredRelatedContent->getDiscussionThreads(),
-					$wikiArticles);
-
-			foreach ($ordered as $sublist) {
-				foreach ($sublist as $item) {
-					if (!empty($item)) {
-						/** @var RelatedContent $item */
-						$content = $item->getContent();
-
-						$items[] = new RecirculationContent( [
-								'index' => count($items),
-								'url' => $content->getUrl(),
-								'thumbnail' => $content->getImage(),
-								'title' => $this->formatTitle($content),
-							] );
-					}
-
-					if (count($items) >= $limit) {
-						break 2;
-					}
-				}
+		$wikiArticles = [];
+		foreach ( $filteredRelatedContent->getWikiArticles() as $article ) {
+			$parsed = parse_url( $article->getContent()->getUrl() );
+			if ( urldecode( $parsed[ 'path' ] ) != $ignore ) {
+				$wikiArticles[] = $article;
 			}
+		}
 
-			return $items;
-		} catch (ApiException $e) {
-			$this->error("error while getting content", [
-				'code' => $e->getCode(),
-				'message' => $e->getMessage(),
-			]);
-			return [];
+		$items = [];
+		$fandomArticles = array_slice( $filteredRelatedContent->getFandomArticles(), 0, $limit - count( $items ), true );
```

Comment:
Forget about negative length here, but I think it's still better concept than 
`array_map(null, ...)` and `break 2`